src/libcore/slice/mod.rs

```diff
@@ -1215,6 +1215,42 @@ macro_rules! iterator {
                 }
                 accum
             }
+
+            #[inline]
+            fn position<P>(&mut self, mut predicate: P) -> Option<usize> where
+                Self: Sized,
+                P: FnMut(Self::Item) -> bool,
+            {
+                // The addition might panic on overflow
```

Comment:
I think this needs either `Add::add` instead of `+` or `#[rustc_inherit_overflow_checks]`.

Also, it's a shame there's no way to "call base" here.  Would it be worth making a `pub(crate)` helper here to avoid duplicating the logic?  Or using specialization to do the `assume` for all `Self: TrustedLen`?

Reply:
I don't think the duplication is a concern as these are really small. Having a helper would probably end up adding more lines than saving.

Good points about the overflow and specialization though. I have to test if the assume survives the abstraction/inlining.

Reply:
I couldn't make it work with TrustedLen 😞  https://godbolt.org/g/bFA3tk

Reply:
Thanks for giving it a try!  Good to see this get in 🙂